### DISCUSSION TITLE
Strange font coloring on certain letters

### DISCUSSION THREAD

**Discussion by bound-variable** [KDE/ghostwriter#688] [General]
After converting a Word .docx file to Markdown via Pandoc, the editing pane in ghostwriter displays altered font colors for certain, seemingly random, letters. It happens with all themes, but the examples are of the Classic theme.

![image](https://user-images.githubusercontent.com/51193876/141004278-ff4d8bdf-a870-453c-b837-b005b8a2d950.png)

![image](https://user-images.githubusercontent.com/51193876/141004316-16f0743f-1580-4de6-8cb5-f05df02bd940.png)

- App ver: 2.02
- OS: Debian 11, Gnome 3.38.5

**Comment by goebbe** [comment]
I have similar issues, sometimes, when working on a text. (App vers 2.02, LinuxMint 19.2, Cinnamon X11) But I couldn't find a way to reproduce reliably. 
Question: Is the rendering o.k. if you restart ghostwriter?

**Reply by bound-variable** [reply]
> I have similar issues, sometimes, when working on a text. (App vers 2.02, LinuxMint 19.2, Cinnamon X11) But I couldn't find a way to reproduce reliably. Question: Is the rendering o.k. if you restart ghostwriter?

No, the rendering was always like that even after restarting. It didn't only occur with converted .docx files, but I can't recall clearly those other instances now.

**Reply by zaqqwerty** [reply]
Can confirm, for me once it happens to a file, restarting does not correct it

**Comment by zaqqwerty** [comment]
I managed to reproduce this, seems to happen when text is highlighted or pasted in.  For example, this text:

```
Test sentence.
Another one,
will this
trigger
the bug?
```

copy into a new file, highlight it and click away, and the letters s in Test and o on Another are greyed out.

On Ubuntu, ghostwriter version "2.0.2+202106281733~ubuntu20.04.1"

**Reply by zaqqwerty** [reply]
Downgraded to the other version available on Ubuntu, `1.8.0-2build1` - the highlighting issue does not occur there, so it seems the bug occurred in some change between 1.8 and 2.0.2

**Reply by zaqqwerty** [reply]
Isolated further with local builds, seems to have been introduced between `v1.8.1` and `2.0.0-rc1`

**Reply by zaqqwerty** [reply]
Found the PR that introduced the bug: https://github.com/wereturtle/ghostwriter/commit/7d1a8b0a65becb02ae191e60eb17d2fcbcfc0f53

**Comment by goebbe** [comment]
This issue has been fixed in Ghostwriter 2.1.2.  
Thank you very much to the developer(s)!
